easybuild/easyblocks/g/gromacs.py

```diff
@@ -203,6 +205,65 @@ def configure_step(self):
                 # to avoid that GROMACS finds and uses a system-wide CUDA compiler
                 self.cfg.update('configopts', "-DGMX_GPU=OFF")
 
+        # CP2K detection
+        # enable CP2K support if CP2K is listed as a dependency
+        # and CP2K support is either explicitly enabled (cp2k = True) or unspecified ('cp2k' not defined)
+        cp2k_root = get_software_root('CP2K')
+        if self.cfg['cp2k'] and not cp2k_root:
+            msg = "CP2K support has been requested but CP2K is not listed as a dependency."
+            raise EasyBuildError(msg)
+        elif cp2k_root and self.cfg['cp2k'] is False:
+            self.log.info('CP2K was found, but compilation without CP2K has been requested.')
+            cp2k_root = None
+
+        if cp2k_root:
+            if LooseVersion(self.version) < LooseVersion('2022'):
+                msg = 'CP2K support is only available for GROMACS 2022 and newer.'
+                raise EasyBuildError(msg)
+            elif LooseVersion(get_software_version('CP2K')) < LooseVersion('8.1'):
+                msg = 'CP2K support in GROMACS requires CP2K version 8.1 or higher.'
+                raise EasyBuildError(msg)
+
+            if not self.cfg['mpi_only']:
+                msg = "GROMACS with CP2K support needs to be built with 'mpi_only = True'"
+                raise EasyBuildError(msg)
+
+            if not os.path.exists(os.path.join(cp2k_root, 'lib', 'libcp2k.a')):
+                msg = 'CP2K needs to be compiled with "library = True".'
+                raise EasyBuildError(msg)
+            if not os.path.exists(os.path.join(cp2k_root, 'lib', 'pkgconfig', 'libcp2k.pc')):
+                msg = "pkg-config is required as a build-dependency for CP2K"
+                raise EasyBuildError(msg)
+            if not get_software_root('pkg-config'):
+                msg = "pkg-config is required as a build-dependency for building GROMACS-CP2K"
+                raise EasyBuildError(msg)
+
+            self.log.info('CP2K support has been enabled.')
+            # Building with CP2K requires static build w/o gmxapi.
+            # https://manual.gromacs.org/documentation/2022/install-guide/index.html#building-with-cp2k-qm-mm-support
+            self.log.info("Building with CP2K QM/MM.")
+            self.cfg['build_shared_libs'] = False
+            self.libext = 'a'
+            cp2k_version = get_software_version('CP2K')
```

Comment:
Move this up above the first use of CP2K version

Reply:
done in 6b1f77d88